declcache.go

```diff
@@ -224,9 +225,23 @@ func build_package(p *build.Package) error {
 		log.Printf("package object: %s", p.PkgObj)
 		log.Printf("package source dir: %s", p.Dir)
 		log.Printf("package source files: %v", p.GoFiles)
+		log.Printf("GOPATH: %v", g_daemon.context.GOPATH)
+		log.Printf("GOROOT: %v", g_daemon.context.GOROOT)
 	}
+	env := os.Environ()
+	for i, v := range env {
+		if strings.HasPrefix(v, "GOPATH=") {
+			env[i] = "GOPATH=" + g_daemon.context.GOPATH
+		} else if strings.HasPrefix(v, "GOROOT=") {
```

Comment:
> it's a fair point. We should ensure both `GOROOT` and `GOPATH` are set.

If you want, you're welcome to use this package to "set" (meaning overwrite existing value, or add if it doesn't exist) env vars of `os.Environ()`.

https://godoc.org/github.com/shurcooL/go/osutil

(See example usage [here](https://gotools.org/github.com/shurcooL/vcsstate?rev=20a6595affd14b7449f1dce9df5a17557480df6b#remotegit-go-L17-L18).)

You can also just take its code if you don't wish to add a dependency.